5-Heroku/client/toDoListApp.js

```diff
@@ -0,0 +1,189 @@
+    const buttonsWrapper = createHtmlElement("div", [classes.buttonsWrapper]);
+    const editButtonElement = createEditButton(taskInfo);
+    const deleteButtonElement = createDeleteButton(taskInfo);
+    buttonsWrapper.appendChild(editButtonElement);
+    buttonsWrapper.appendChild(deleteButtonElement);
+    return buttonsWrapper;
+  }
+
+  function createEditButton(taskInfo) {
+    const editButtonElement = createButtonElement(
+      [classes.editTaskInfo, classes.buttonStyle],
+      "click",
+      editElementButtonHandler,
+      taskInfo
+    );
+    editButtonElement.textContent = "edit";
+    return editButtonElement;
+  }
+
+  function createDeleteButton(taskInfo) {
+    const deleteButtonElement = createButtonElement(
+      [classes.deleteTask, classes.buttonStyle],
+      "click",
+      deleteElementButtonHandler,
+      taskInfo.id
+    );
+    deleteButtonElement.textContent = "delete";
+    return deleteButtonElement;
+  }
+
+  function createTaskInfoElement(title) {
+    const taskTitleElement = createHtmlElement("h3", [classes.toDoTaskTitle]);
+    taskTitleElement.id = "task-title";
+    taskTitleElement.textContent = title;
+    return taskTitleElement;
+  }
+
+  async function createTask(title,id='', isNewTask = true) {
+    let taskId = id;
+    if (isNewTask){
+      taskId = (await addTask(title)).id;
+    }
+    const taskInfo = {
+      id: taskId,
+      title: title,
+      isEditMode: false,
+      isDone: false,
+      
+    };
+    const toDoTaskElement = createHtmlElement("li", [classes.toDoTasks]);
+    toDoTaskElement.id = taskInfo.id;
+    const checkBoxElement = createTaskCheckBoxElement(taskInfo);
+    toDoTaskElement.appendChild(checkBoxElement);
+    const taskInfoElement = createTaskInfoElement(taskInfo.title);
+    toDoTaskElement.appendChild(taskInfoElement);
+    const buttonsWrapper = createButtonElements(taskInfo);
+    toDoTaskElement.appendChild(buttonsWrapper);
+    let ulElement = document.getElementById("to-do-list-container");
+    ulElement.appendChild(toDoTaskElement);
+  }
+
+  function createTaskCheckBoxElement(taskInfo) {
+    const checkBoxElement = createHtmlElement("input");
+    checkBoxElement.type = "checkbox";
+    checkBoxElement.id = "task-checkbox";
+    checkBoxElement.addEventListener("click", (e) => {
+      markTaskAsDone(e, taskInfo);
+    });
+    return checkBoxElement;
+  }
+
+  function createHtmlElement(tagName, className = []) {
+    const htmlElement = document.createElement(tagName);
+    htmlElement.classList.add(...className);
+    return htmlElement;
+  }
+
+  async function deleteElementButtonHandler(event, id) {
+    const task = document.getElementById(id);
+    task.remove();
+    await deleteTask(id);
+  }
+
+  function editElementButtonHandler(event, taskInfo) {
+    if (!taskInfo.isEditMode) {
+      editElementTitle(taskInfo);
+    } else {
+      setElementTitle(taskInfo);
+    }
+    taskInfo.isEditMode = !taskInfo.isEditMode;
```

Comment:
i don't think that `isEditMode` should be part of the data object that you're sending to the server. edit is a UI concern and it should stay there, you could either remove this property before sending or, the preferred way imo, store the editable items in a different object in the client side.
for instance, you would have a `editedItems = []` that holds all of the currently edited items